src/cmd_all/src/lib.rs

```diff
@@ -16,3 +16,27 @@
 
 pub mod playground;
 pub use playground::*;
+
+lazy_static::lazy_static! {
+    static ref VERSION: String = {
+        let version = clap::crate_version!();
+        // VERGEN_GIT_SHA is provided by the build script. It will trigger rebuild
+        // for each commit, so we only use it for the final binary.
+        let vergen_git_sha = env!("VERGEN_GIT_SHA");
+
+        // GIT_SHA is a normal environment variable. It's risingwave_common::GIT_SHA
+        // and is used in logs/version queries.
+        //
+        // Usually it's only provided by docker/binary releases (including nightly builds).
+        // We check it is the same as VERGEN_GIT_SHA when it's present.
+        if let Some(git_sha) = option_env!("GIT_SHA") {
+            assert!(git_sha.to_lowercase().contains(vergen_git_sha), "GIT_SHA ({git_sha}) mismatches VERGEN_GIT_SHA ({vergen_git_sha})");
```

Comment:
Can we make this a compile-time check?

Reply:
<img width="808" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/37948597/05c1cd89-0357-4e53-9142-7ad5e2173835">